src/htm/algorithms/Connections.cpp

```diff
@@ -519,6 +519,9 @@ void Connections::adaptSegment(const Segment segment,
     }
   }
 
+  //balance synapses using competition on dendrite
+  synapseCompetition(segment, 8, 11); //FIXME derive these numbers
```

Comment:
this is the only big remaining issue - how to choose good values for these? 
Any tips @ctrl-z-9000-times ?